File: 1146-GreatestCommonDivisorOfStrings/1146-GreatestCommonDivisorOfStrings.py
# Last updated: 08/05/2026, 09:28:35
class Solution(object):
    def gcdOfStrings(self, str1, str2):
        """
        :type str1: str
        :type str2: str
        :rtype: str
        """
        strx=""
        if len(str1)<=len(str2):
            maxL =0
            for i in range(len(str1)):
                L2 = str2.count(str1[:i+1])
                L1= str1.count(str1[:i+1])
                if (((i+1)*L2) == len(str2) and ((i+1)*L1) == len(str1)):
                    maxL = i+1
                    strx= str1[:i+1]

        else:
            maxL =0
            for i in range(len(str2)):
                L2 = str2.count(str2[:i+1])
                L1= str1.count(str2[:i+1])
                if (((i+1)*L2) == len(str2) and ((i+1)*L1) == len(str1)):
                    maxL = i+1
                    strx= str2[:i+1]

        return strx

File: 1146-GreatestCommonDivisorOfStrings/1146-GreatestCommonDivisorOfStrings.cpp
// Last updated: 08/05/2026, 09:28:31
class Solution {
public:
    string gcdOfStrings(string str1, string str2) {
        if(str1+str2!=str2+str1){
            return "";
        }
        
        auto GCD= []( int len1, int len2){
            while(len2!=0) {
            int rem=len1 % len2;
            len1= len2;
            len2 = rem;
            
            }
            return len1;
        };
        int lenGCD= GCD(str2.length(),str1.length());

        return str1.substr(0,lenGCD);

    }
};